### PR TITLE
[full-ci][tests-only]Refactor sharing related tests and add notToImplementOnOCIS tag

### LINKED ISSUE #2317: Sharing Test for oCIS
I had a look at why `apiVersions/fileVersions.feature:276` was failing with oCIS.

I got the test working with a few changes.
First a fix for version listing: https://github.com/cs3org/reva/pull/1911
And then I had to edit the test a bit.

```Gherkin
Scenario: sharer can restore a file inside a group shared folder modified by sharee
    Given user "Brian" has been created with default attributes and without skeleton files
    And user "Carol" has been created with default attributes and without skeleton files
    And group "grp1" has been created
    And user "Brian" has been added to group "grp1"
    And user "Carol" has been added to group "grp1"
    And user "Alice" has created folder "/sharingfolder"
    And user "Alice" has shared folder "/sharingfolder" with group "grp1"
    And user "Alice" has uploaded file with content "First content" to "/sharingfolder/sharefile.txt"
    And user "Brian" accepts share "/sharingfolder" offered by user "Alice" using the sharing API
    And user "Brian" has uploaded file with content "Second content" to "/Shares/sharingfolder/sharefile.txt"
    And user "Carol" accepts share "/sharingfolder" offered by user "Alice" using the sharing API
    And user "Carol" has uploaded file with content "Third content" to "/Shares/sharingfolder/sharefile.txt"
    When user "Alice" restores version index "2" of file "/sharingfolder/sharefile.txt" using the WebDAV API
    Then the HTTP status code should be "204"
    And the content of file "/sharingfolder/sharefile.txt" for user "Alice" should be "First content"
    And the content of file "/Shares/sharingfolder/sharefile.txt" for user "Brian" should be "First content"
    And the content of file "/Shares/sharingfolder/sharefile.txt" for user "Carol" should be "First content"
```

Before the share recipients upload to the share I added steps for them to accept the share first.
Also I updated the paths for the share recipients to include the `/Shares` folder.

This means that we probably need to have two versions of the test. The original version for oC10 and this version for oCIS.

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
This PR refactors tests related to sharing.
Changes made in this PR:
- Tag scenarios that share to root folder with `@notToImplementOnOCIS`
- Add a extra blank line between the scenarios
- Add test scenarios that share to root folder 

## Related Issue

- Fixes : https://github.com/owncloud/ocis/issues/2317

## Motivation and Context
- Tests scenarios that share directly to root folder should be tagged with `@notToImplementOnOCIS` as the behavior in  OCIS is different. Some tests weren't tagged  `@notToImplementOnOCIS` so it was necessary to tag them. 
- We have this new convention to add a extra blank line between the scenarios so that we can add issue or any other tag without disrupting the line numbers of other test scenarios. As I was working on these files and the convention wasn't followed I added the extra lines.

## How Has This Been Tested?

- locally
- ci
- https://github.com/owncloud/ocis/pull/2327

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by swoichha — COMMENTED** [review]
lgtm :+1:

File: tests/acceptance/features/apiVersions/fileVersions.feature [hunk 1]
```diff
@@ -355,7 +356,7 @@ Feature: dav-versions
       | old         | Brian |
       | new         | Brian |
 
-  @skipOnOcV10.3.0 @files_sharing-app-required @issue-ocis-1238
+  @skipOnOcV10.3.0 @files_sharing-app-required @issue-ocis-1238 @notToImplementOnOCIS
```
Comment (phil-davis) [inline tests/acceptance/features/apiVersions/fileVersions.feature:359]: Note: the version of this scenario for "sharing to shares" already exists in `fileVersionsSharingToShares.feature` - good

File: tests/acceptance/features/apiVersions/fileVersions.feature [hunk 2]
```diff
@@ -365,7 +366,7 @@ Feature: dav-versions
     Then the HTTP status code should be "404"
     And the value of the item "//s:exception" in the response about user "Alice" should be "Sabre\DAV\Exception\NotFound"
 
-  @skipOnStorage:ceph @files_primary_s3-issue-161 @files_sharing-app-required
+  @skipOnStorage:ceph @files_primary_s3-issue-161 @files_sharing-app-required @notToImplementOnOCIS
```
Comment (phil-davis) [inline tests/acceptance/features/apiVersions/fileVersions.feature:369]: Note: the version of this scenario for "sharing to shares" already exists in `fileVersionsSharingToShares.feature` - good

**Review by phil-davis — APPROVED** [review]
LGTM

**Pull request merged**